src/tools/fuzzing.h

```diff
@@ -934,8 +934,14 @@ class TranslateToFuzzReader {
     // Pick a chance to fuzz the contents of a function.
     const int RESOLUTION = 10;
     auto chance = upTo(RESOLUTION + 1);
+    // Do not iterate on wasm.functions itself, as we may add to it as we go
+    // through the functions (make() can add new functions to implement a
+    // RefFunc).
+    std::vector<Function*> functions;
     for (auto& ref : wasm.functions) {
-      auto* func = ref.get();
+      functions.push_back(ref.get());
+    }
+    for (auto* func : functions) {
```

Comment:
Oh, good point, thanks! Done.

Also I think we can just iterate with a dynamic check on `size()`. It's not bad to iterate on the new functions, in fact we might as well, to avoid bugs in the future.